ci/functional/test_main_prep_node.sh

```diff
@@ -2,33 +2,226 @@
 
 set -eux
 
-for i in 0 1; do
+: "${FIRST_NODE:=}"
+: "${OPERATIONS_EMAIL:=}"
+
+result=0
+mail_message=''
+nl="
+"
+
+function do_mail {
+    if [ -z "$mail_message" ]; then
+        return
+    fi
+    set +x
+    if [ -z "$OPERATIONS_EMAIL" ]; then
+        echo "$mail_message"
+        return
+    fi
+    # shellcheck disable=SC2059
+    mail -s "Hardware check failed after reboot!" \
+         -r "$HOSTNAME"@intel.com "$OPERATIONS_EMAIL" \
+         <<< "$mail_message"
+    set -x
+}
+
+if ! command -v lspci; then
+    if command -v dnf; then
+       dnf -y install pciutils
+    else
+       echo "pciutils not installed, can not test for Infiniband devices"
+    fi
+fi
```

Comment:
Don't we already have a package list?  It seems odd to be adding another location in the code paths to download/install packages if we know it's always going to be needed.